src/Generator/RequestGenerator.php

```diff
@@ -129,6 +135,8 @@ protected function generateConstructor(Request $request): ?ClassMethod
             return null;
         }
 
+        $params = $this->sortParameters(...$params);
```

Comment:
just wondering, any specific reason for this? and why didn't it break any tests, all yamls are sorted already? 😄

Reply:
Yep this is because without this sorting stuff, we would have things like this (depending on the order in the open api spec):
```php
public function __construct(string $param1 = 'default_value', string $param2) {}
```

It didn't break any test because the existing yamls are not using a combo "required parameter + default value" on a request. I handled the case under the PatchResource functional test.